### PR TITLE
Add querying and removal of test results.

### PR DESCRIPTION
**Pull request by Khubajsn**
1. obrazek
2. `CommandExtensions.GetCommandFormat` has been updated to work properly with command groups.
3. Added `IEmbedMetadata` + `EmbedMetadataExtensions`, which allows for concealing metadata in embeds. This is needed when processing embed reactions/buttons, so that it's possible to determine which module is responsible for editing the embed, if any. The HelpPageEmbed and related classes have been updated to use IEmbedMetadata as well.

    The `embedBuilder.WithMetadata` extension method will serialize the metadata into a URL query string-like format (such as `_k=HelpEmbed&PageNumber=1&PageCount=10`) and store it in the embed, somehow. The following destination selection logic is used:
    
    1. If the embed has an author with an avatar URL, that avatar URL is edited so that it's fragment portion (starting with `#`) includes the metadata.
    2. If the embed has an image URL set, that URL is edited in the same manner.
    3. The metadata are stored into the embed's footer.
    
    Later on, the `Embed.TryParseMetadata` extension can be used to parse/recover the data. The same logic is used to determine where to look as above.

4. Added an `Entity` subdirectory/namespace inside `Inkluzitron.Data`, so that the entity classes are kept separate from the not-so-simple `DataContext`.
5. Moved test-related modules into their own subdirectory/namespace.
6. Moved `Inkluzitron.Settings` to `Inkluzitron.Models.Settings`

### CODE REVIEW

File: src/Inkluzitron/Data/Entities/QuizResult.cs [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
     public class QuizResult
     {
-        public ulong ResultId { get; set; }
+        public long ResultId { get; set; }
         public DateTime SubmittedAt { get; set; }
         public ulong SubmittedById { get; set; }
         public string SubmittedByName { get; set; }
         public List<QuizItem> Items { get; set; } = new List<QuizItem>();
-    }
+
```
Comment (Misha12) [inline src/Inkluzitron/Data/Entities/QuizResult.cs:13]: Ty prázdné řádky tu musí být?

File: src/Inkluzitron/Models/Settings/BdsmTestOrgSettings.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        public string NoTraitsToReportMessage { get; }
+        public string InvalidVerboseModeUsage { get; }
+        public string BadFilterQueryMessage { get; }
+        public IReadOnlySet<string> TraitList { get; }
+        public string TestLinkUrl { get; }
+
+        public BdsmTestOrgSettings(IConfiguration config)
+        {
+            const string SectionName = "BdsmTestOrgQuizModule";
+            string GetRequiredConfig(string key)
```
Comment (Misha12) [inline src/Inkluzitron/Models/Settings/BdsmTestOrgSettings.cs:29]: Tahle metoda by se krásně dala napsat jako extension metoda nad typem IConfiguration `string GetRequiredValue(this IConfiguration configuration, string key);`

File: src/Inkluzitron/Modules/BdsmTestOrg/BdsmModule.cs [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        public async Task ShowUserResultsAsync()
+        {
+            var authorId = Context.Message.Author.Id;
+            var quizResultsOfUser = DbContext.BdsmTestOrgQuizResults
+                .Include(x => x.Items)
+                .Where(x => x.SubmittedById == authorId);
+
+            var pageCount = await quizResultsOfUser.CountAsync();
+            var mostRecentResult = await quizResultsOfUser.OrderByDescending(r => r.SubmittedAt)
+                .Take(1)
```
Comment (Misha12) [inline src/Inkluzitron/Modules/BdsmTestOrg/BdsmModule.cs:57]:
```suggestion
                .FirstOrDefaultAsync();
```

File: src/Inkluzitron/Modules/BdsmTestOrg/BdsmModule.cs [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+                .FirstOrDefaultAsync();
```
Comment (Misha12) [inline src/Inkluzitron/Modules/BdsmTestOrg/BdsmModule.cs:58]:
```suggestion
```

File: src/Inkluzitron/Modules/BdsmTestOrg/BdsmModule.cs [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+            {
+                await ReplyAsync(Settings.InvalidFormatMessage);
+                return;
+            }
+
+            var testResultMatch = testResultMatches.Single();
+            var testResultItems = testResultMatch.Groups["results"].Value;
+            var testResultLink = testResultMatch.Groups["link"].Value;
+
+            if (await DbContext.BdsmTestOrgQuizResults.AsAsyncEnumerable().AnyAsync(r => r.Link == testResultLink))
```
Comment (Misha12) [inline src/Inkluzitron/Modules/BdsmTestOrg/BdsmModule.cs:175]:
```suggestion
            if (await DbContext.BdsmTestOrgQuizResults.AsQueryable().AnyAsync(r => r.Link == testResultLink))
```

Pokud se nepletu, tak `.AsAsyncEnumerable` zařízne SQL dotaz, tudíž se SQL vyhodnotí, že se mají stáhnout všechna data.

File: src/Inkluzitron/Contracts/IReactionHandler.cs [hunk 6]
```diff
@@ -5,6 +5,6 @@ namespace Inkluzitron.Contracts
 {
     public interface IReactionHandler
     {
-        Task<bool> Handle(IUserMessage message, IEmote reaction, IUser user);
+        Task<bool> Handle(IUserMessage message, IEmote reaction, IUser user, IUser botUser);
```
Comment (Misha12) [inline src/Inkluzitron/Contracts/IReactionHandler.cs:8]: Zřejmě to bude na více místech, takto dám jako comment. 
Mohli bychom držet zavedený standard, který myslím, že má i samotnej .NET. A to, že asynchronní metody končí na Async. Takže tady by se to jmenovalo `HandleAsync`.

**Review by Misha12 — CHANGES_REQUESTED** [review]

**Review by Misha12 — APPROVED** [review]

**Review by Misha12 — APPROVED** [review]

**Review by janch32 — APPROVED** [review]
Budu věřit Michalovi 😀

**Pull request merged**